pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -562,7 +562,7 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 
 		skuI, err := s.resourcesSkus.Get(ctx, skuSpec)
 		if err != nil {
-			return fmt.Errorf("failed to find sku %s", s.scope.MachineConfig.VMSize)
+			return machinecontroller.InvalidMachineConfiguration("failed to fetch instance type information from azure: %w", err)
```

Comment:
Could this be a transient error? What machine configuration leads into this? Is there a way to be more precise here, eg, if it returns a not found error vs something else?